third_party/imgui/.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,42 @@
+## How to create an Issue
```

Comment:
This looks like a cool library, but would you mind adding it as a git submodule? It's what we're doing elsewhere (e.g. for ARSDK and msp) and does make things a bit tidier IMHO.

Reply:
Cool. I do have a makefile for the ARSDK that checks out the submodule and builds it if you'd like an example.

Reply:
I have attempted to set this up - could you check it works for you (building the new rothamsted_reconstruction repro is probably a good test)

Reply:
Yep, it builds fine for me.

You could also retrieve the git submodule with the makefile:
``
submodule:
    $(shell git submodule update --init imgui)
``

Then make ``%.o`` depend on ``submodule``.

Reply:
...sorry, misread your comment. The submodule builds ok for me. I'll test the rothamsted_reconstruction repo now.

Reply:
Right, so it did build for me, but I had to change the includes to read:
``
// BoB robotics third-party includes
#include "third_party/imgui/imgui/imgui.h"
#include "third_party/imgui/imgui/imgui_internal.h"
#include "third_party/imgui/imgui/examples/imgui_impl_glfw.h"
#include "third_party/imgui/imgui/examples/imgui_impl_opengl2.h"
``

Reply:
The includes in the submodule or those in the rothamsted reconstruction repo (which I definitely failed to push...)?

Reply:
The submodule wasn't being checked out first for me (my mistake!). I've pushed a fix. Other than that I think we're done!